x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/data_tier_allocation/data_tier_allocation.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSwitch, EuiText, EuiFormRow } from '@elastic/eui';
+
+import { PhaseWithAllocationAction } from '../../../../../../common/types';
+import { AdvancedSectionLayout } from '../advanced_section_layout';
+import { NodeAllocation } from './node_allocation';
+import { SharedProps } from './types';
+
+import './data_tier_allocation.scss';
+import { LearnMoreLink } from '../learn_more_link';
+
+const learnMoreLink = (
+  <LearnMoreLink
+    text={
+      <FormattedMessage
+        id="xpack.indexLifecycleMgmt.editPolicy.learnAboutShardAllocationLink"
+        defaultMessage="Learn about shard allocation"
+      />
+    }
+    docPath="modules-cluster.html#cluster-shard-allocation-settings"
+  />
+);
+
+const i18nTexts = {
+  useDataTierAllocation: i18n.translate(
+    'xpack.indexLifecycleMgmt.editPolicy.common.dataTierAllocation.useDataTierSwitchLabel',
+    { defaultMessage: 'Enable data tier allocation' }
+  ),
+  useCustomDataTierAllocation: i18n.translate(
+    'xpack.indexLifecycleMgmt.editPolicy.common.dataTierAllocation.useCustomDataTierSwitchLabel',
+    { defaultMessage: 'Use custom data tier allocation' }
```

Comment:
I like "Custom attribute-based allocation", but I'm not sure replacing "Enable data tier allocation" with "Automatic role-based allocation" works well with this because the intention is to introduce custom allocation as a specialised version of data tier allocation. Could something like this work?

<img width="455" alt="Screenshot 2020-09-09 at 16 02 59" src="https://user-images.githubusercontent.com/8155004/92608666-f44e5c80-f2b5-11ea-97c4-91271b3bf3fe.png">